src/main/java/org/jboss/xavier/integrations/rbac/RBACService.java

```diff
@@ -0,0 +1,265 @@
+            String value = attributeFilter.getValue();
+
+            if (operation.equals("equal") && value != null) {
+                result.add(value);
+            } else if (operation.equals("in") && value != null) {
+                result.addAll(Arrays.asList(value.split(",")));
+            }
+        }
+
+        return result;
+    }
+
+    /***
+     * Process acls to determine capabilities. Empty resourceDefinitions means all access ('*')
+     *
+     * @return List of resourceDefinition values.
+     * E.g. Given:
+     * [
+     *     {
+     *         "permission": "migration-analytics:resource1:read",
+     *         "resourceDefinitions": []
+     *     },
+     *     {
+     *         "permission": "migration-analytics:resource2:write",
+     *         "resourceDefinitions": [
+     *             {
+     *                 "attributeFilter": {
+     *                     "key": "key1",
+     *                     "operation": "equal",
+     *                     "value": "1"
+     *                 }
+     *             },
+     *             {
+     *                 "attributeFilter": {
+     *                     "key": "key2",
+     *                     "operation": "in",
+     *                     "value": "2,3,4"
+     *                 }
+     *             }
+     *         ]
+     *     }
+     * ]
+     *
+     * Will return:
+     * {
+     *     "resource1": [
+     *         {
+     *             "operation": "read",
+     *             "resources": [
+     *                 "*"
+     *             ]
+     *         }
+     *     ],
+     *     "resource2": [
+     *         {
+     *             "operation": "write",
+     *             "resources": [
+     *                 "1",
+     *                 "2",
+     *                 "3",
+     *                 "4"
+     *             ]
+     *         }
+     *     ]
+     * }
+     * */
+    public static Map<String, List<AclData>> processAcls(List<Acl> acls) {
+        Map<String, List<AclData>> access = new HashMap<>();
+        for (Acl acl : acls) {
+            String permission = acl.getPermission();
+            List<Acl.ResourceDefinition> resourceDefinitions = acl.getResourceDefinitions();
+
+            // extract permission_data
+            String[] permComponents = permission.split(":");
+            if (permComponents.length != 3) {
+                throw new IllegalStateException("Invalid permission definition permission:" + permission);
+            }
+
+            String resourceType = permComponents[1];
+            String operation = permComponents[2];
+
+            List<String> resources = extractResourceDefinitions(resourceDefinitions);
+            AclData aclData = new AclData(operation, resources);
+
+            if (!access.containsKey(resourceType)) {
+                access.put(resourceType, new ArrayList<>());
+            }
+            access.get(resourceType).add(aclData);
```

Comment:
We can rely upon Map's `computeIfAbsent` method to manage the "missing key case"
```suggestion
            access.computeIfAbsent(resourceType, resource -> new ArrayList<>()).add(aclData);
```

Reply:
Thanks @mrizzi